projects/storefrontlib/cms-components/product/carousel/product-carousel/product-carousel.component.ts

```diff
@@ -15,7 +15,11 @@ import { CmsComponentData } from '../../../../cms-structure/page/model/cms-compo
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ProductCarouselComponent {
-  protected readonly PRODUCT_SCOPE = ProductScope.LIST;
+  protected readonly PRODUCT_SCOPE = [
+    ProductScope.LIST,
+    ProductScope.PRICE,
+    ProductScope.DETAILS,
```

Comment:
Is there a way to not load the whole product details in order to fix the product stock issue?  
Reasoning: The purpose of the scopes were to delay the moment we need to load the whole product details, usually when the user hits the product details page.

Reply:
I'm not sure - these are the only scopes available for fetching products, at least written in SPA code:

- LIST = 'list'
- DETAILS = 'details'
- ATTRIBUTES = 'attributes'
- VARIANTS = 'variants'
- PRICE = 'price'

DETAILS seems to be the only adequate one here.

Reply:
Which fields are required for the out of stock check to work properly?
Perhaps there is a way to get those fields without all the product details. I don't know exactly what or how, this is a suggestion.

Perhaps the add to cart button on the search results page and/or category page can be examined for how they do it.  I don't know if they react to out of stock status and it they do, perhaps we can see if they work with less data than all the product details.

Reply:
At a high level, a STOCK scope sounds optimal since it's probably a common use case that the product carousel would be used on the home page.
I don't know how involving this modification can be though.  I never developed a scope.

In the end, I'm not 100% against using details, but if there is a way to be optimal with a reasonable/small amount of effort, it might be worth it since this component is likely used on the home page.

Reply:
Turned out to be pretty simple after all - PR updated.